lib/ansible/module_utils/basic.py

```diff
@@ -480,9 +480,11 @@ def tmpdir(self):
             if basedir is not None and not os.path.exists(basedir):
                 try:
                     os.makedirs(basedir, mode=0o700)
-                except (OSError, IOError) as e:
-                    self.warn("Unable to use %s as temporary directory, "
-                              "failing back to system: %s" % (basedir, to_native(e)))
+                except OSError as ex:
+                    self.error_as_warning(
+                        f"Unable to use {basedir!r} as temporary directory, falling back to system default.",
```

Comment:
When splitting args like this across multiple lines, I think using kwargs where possible improves readability:

```suggestion
                        msg=f"Unable to use {basedir!r} as temporary directory, falling back to system default.",
```